### DISCUSSION TITLE
Why are grim screenshots unreliable?

### DISCUSSION THREAD

**Discussion by txtyash** [hyprwm/Hyprland#2615] [Q&A] [answered]
I'm able to take screenshots with: 
```
grim -g "0,0 1920x1080" $(xdg-user-dir)/screenshots/full/$(date +'%F-%T.png')
```
but when I try to make a binding to copy them to `wl-copy` then it just doesn't work. There was only one example in the outdated `grim` repo:
```
grim - | wl-copy`
```
and when paired with my above command:
```
grim - -g "0,0 1920x1080" $(xdg-user-dir)/screenshots/full/$(date +'%F-%T.png') | wl-copy
```
It fails to run.

There aren't any other reliable options to take screenshots in wayland either.

**Comment by MightyPlaza** [comment, marked as answer]
```
exec-once =wl-copy

bind = $mainMod SHIFT, S, exec, grim -g "$(slurp)" - | tee >(wl-copy -t image/png) > $HOME"/Pictures/Screenshots/Screenshot"$(date +"%Y-%m-%d_%H-%M-%S")".png" 
```
you're misssing the type

**Comment by txtyash** [comment]
@MightyPlaza what do u use for screenshots of active/focused window?

**Reply by MightyPlaza** [reply]
I don't but just change the slurp part of the command above with the parsing of hyprctl that's on the arch wiki.
https://wiki.archlinux.org/title/Screen_capture#Wayland

```
hyprctl -j activewindow | jq -r '"\(.at[0]),\(.at[1]) \(.size[0])x\(.size[1])"'
```
